app/controllers/katello/api/v2/activation_keys_controller.rb

```diff
@@ -87,16 +87,16 @@ def copy
       @new_activation_key.user = current_user
       sync_task(::Actions::Katello::ActivationKey::Create, @new_activation_key)
       @new_activation_key.reload
-      @new_activation_key.update_attributes!(
-                                             :service_level => @activation_key.service_level,
-                                             :release_version => @activation_key.release_version,
-                                             :auto_attach => @activation_key.auto_attach
-                                            )
+      sync_task(::Actions::Katello::ActivationKey::Update, @new_activation_key,
+                  :service_level   => @activation_key.service_level,
+                  :release_version => @activation_key.release_version,
+                  :auto_attach     => @activation_key.auto_attach
+               )
       @activation_key.content_overrides.each do |content|
         @new_activation_key.set_content_override(content['contentLabel'], content[:name], content[:value])
       end
-      @activation_key.get_key_pools.each do |pool|
-        @new_activation_key.subscribe(pool[:id], pool[:amount])
+      @activation_key.subscriptions.each do |subscription|
+        @new_activation_key.subscribe(subscription[:id])
```

Comment:
Does this make calls to Candlepin? If so, we ought to open an issue to dynflowize this.